### PR TITLE
Sorcerer trait adaptation

### PR DESCRIPTION
**Pull request by MemeWizard**
#### Changelog:
- Added sorcerer trait placeholders (currently do nothing) from ck2 version ( Added 8 traits - 3 good inheritable, 3 bad inheritable, 1 good noninheritable, 1 bad noninheritable )
- Spawned witches, debutantes and warriors can appear with this trait
- Added sorcerer trait to level 4 blood legacy perk
- Added sorcerer trait as requirement to strengthen_bloodline_decision
- Reincarnation also can give sorcerer traits

#### Dev changelog:
- added `this_is_magical_society_trigger` (not active) to increase value of characters with sorcerer traits in elections.

# How to test:

- Check all of the above
- Check that ruler transition (death screen) localization related to new traits works.
- Proofread localization

### CODE REVIEW

File: common/traits/wc_genetic_traits.txt [hunk 1]
```diff
@@ -0,0 +1,220 @@
+﻿magic_bad_1 = {
+	index = 2500
+	opposites = {
+		magic_bad_2 #To prevent multiple tiers of the same congenital trait
+		magic_bad_3
```
[the remark this answers is not in the document]
Comment (MemeWizard) [reply common/traits/wc_genetic_traits.txt:5]: Good catch

File: common/customizable_localization/00_ruler_transition_loc.txt [hunk 2]
```diff
@@ -367,6 +367,16 @@ DeathMessageOutro = {
...
+	# Warcraft
+	text = {
+		trigger = { has_trait = magic_good }
+		localization_key = SUCCESSION_EVENT_WINDOW_DEAD_OUTRO_WARCRAFT_MAGIC_GOOD
+	}
+	text = {
+		trigger = { has_trait = magic_bad }
+		localization_key = SUCCESSION_EVENT_WINDOW_DEAD_OUTRO_WARCRAFT_MAGIC_BAD
+	}
+
```
Comment (zumbak04) [inline common/customizable_localization/00_ruler_transition_loc.txt:379]: In vanilla, non-inheritable traits are included here.

File: events/courtier_guest_management_events/courtier_guest_management_events.txt [hunk 3]
```diff
@@ -828,6 +828,10 @@ scripted_trigger courtier_guest_management_3001_interesting_trigger = {
 		has_trait = intellect_good
```
Comment (zumbak04) [inline events/courtier_guest_management_events/courtier_guest_management_events.txt:828]: Looks like this needs adaptation.
<details>
<summary>Click to expand</summary>

image

</details>
I would add scripted triggers here.

File: common/scripted_modifiers/00_elective_successions_scripted_modifiers.txt [hunk 4]
```diff
@@ -1670,6 +1670,42 @@ elector_voting_pattern_traits_feudal_elective_modifier = {
 			}
 		}
 	}
+
+	# Warcraft
+	modifier = {
+		desc = wc_tooltip_feudal_elector_vote_magical
```
Comment (zumbak04) [inline common/scripted_modifiers/00_elective_successions_scripted_modifiers.txt:1676]: We should negative modifier.

File: common/scripted_character_templates/00_soldier_character_templates.txt [hunk 5]
```diff
@@ -177,6 +177,13 @@ new_warrior_character = {
 		physique_good_1 = {}
 		physique_good_2 = {}
 		physique_good_3 = {}
+
+		# Warcraft
+		magic_bad_1 = {}
+		magic_bad_2 = {}
+		magic_bad_3 = {}
+		magic_good_1 = {}
```
Comment (zumbak04) [inline common/scripted_character_templates/00_soldier_character_templates.txt:185]: Shouldn't you get non-inheritable versions too like in vanilla?

File: common/customizable_localization/00_ruler_transition_loc.txt [hunk 6]
```diff
@@ -367,6 +367,16 @@ DeathMessageOutro = {
 		}
```
Comment (zumbak04) [inline common/customizable_localization/00_ruler_transition_loc.txt:367]: Mages like cats, that's the lore.
<details>
<summary>Click to expand</summary>

image

</details>

File: events/courtier_guest_management_events/courtier_guest_management_events.txt (hunk 3 above)
Comment (zumbak04) [inline events/courtier_guest_management_events/courtier_guest_management_events.txt:828]: Guests get interesting traits here and magic traits aren't included.
<details>
<summary>Click to expand</summary>

image

</details>

File: common/customizable_localization/00_ruler_transition_loc.txt (hunk 2 above)
Comment (MemeWizard) [reply common/customizable_localization/00_ruler_transition_loc.txt:379]: > In vanilla, non-inheritable traits are included here.

Can't confirm?

<details>
<summary>Click to expand</summary>

image

</details>

Comment (zumbak04) [reply common/customizable_localization/00_ruler_transition_loc.txt:379]: Nvm.

**Review by zumbak04 — APPROVED** [review]
LGTM!

**Comment by ulricee** [comment]
Seems to work nicely!

File: localization/english/wc_general_l_english.yml [hunk 7]
```diff
@@ -34,4 +34,18 @@
...
+ SUCCESSION_EVENT_WINDOW_DEAD_OUTRO_WARCRAFT_MAGIC_GOOD:0 "Although [ROOT.Char.GetSheHe] was a powerful magi, [ROOT.Char.GetHerHis] abilities did not grant [ROOT.Char.GetHerHim] immortality."
+ SUCCESSION_EVENT_WINDOW_DEAD_OUTRO_WARCRAFT_MAGIC_BAD:0 "Due to an unfortunate lack of potential, [ROOT.Char.GetHerHis] life has had a lot less magic than on average, if any at all."
+ SUCCESSION_EVENT_WINDOW_HEIR_WARCRAFT_MAGIC_GOOD:0 "A [ROOT.Char.GetWomanMan] with a natural affinity for magic, [ROOT.Char.GetHerHis] rule is expected to be full of it."
+ SUCCESSION_EVENT_WINDOW_HEIR_WARCRAFT_MAGIC_BAD:0 "A [ROOT.Char.GetWomanMan] without a hint of magic, [ROOT.Char.GetHerHis] rule is expected to be devoid of miracles."
+
+ child_personality.9900.magic_good_noninheritable.desc:0 "magical,"
+ child_personality.9900.magic_bad_noninheritable.desc:0 "magicless,"
+
+ wc_tooltip_feudal_elector_vote_magical:0 "[candidate.GetSheHe|U] is a potent sorcerer: $VALUE|=+0$"
+ wc_tooltip_feudal_elector_vote_not_magical:0 "[candidate.GetSheHe|U] is a hopeless as magican: $VALUE|=+0$"
```
Comment (TheGroxEmpire) [inline localization/english/wc_general_l_english.yml:51]:
```suggestion
 wc_tooltip_feudal_elector_vote_not_magical:0 "[candidate.GetSheHe|U] is a hopeless magician: $VALUE|=+0$"
```

**Review by TheGroxEmpire — COMMENTED** [review]
I couldn't test the guest with good magical trait notification mechanics. Spamming invite Knight for some time and I could not get a knight with magical trait also. I reckon the chances are so low. Do you think there's a better way to test it? Like some event ID maybe?
But everything else seems to work out fine.
Texts are also looking good for me.

**Comment by MemeWizard** [comment]
@TheGroxEmpire, oh, sorry, I think I remembered it wrong when wrote the PR description. There is no notification about any genetic traits, vanilla, or added by the mod.
It actually was a small text next to a character in the guest tab in the court window
<details>
<summary>Click to expand</summary>

image

</details>

**Review by TheGroxEmpire — COMMENTED** [review]
- This should be "**_An_** Otherwordly woman"
<details>

image
</details>

Vanilla example of it:
<details>

image
</details>

File: localization/english/wc_general_l_english.yml [hunk 8]
```diff
@@ -34,4 +34,18 @@
...
\ No newline at end of file
+ martial_strategy.2001.Mennar.desc:0 "#F The Battle of Mennar is one of the many fights of the War of the Ancients. Instead of regrouping outside the city to join in the counter-assault, the mages of Mennar stayed behind, erecting magical barriers and turning their once-quiet home into a fortress. Refugees streamed to the Mennar Academy in the first terrible days of the invasion, and survivors rallied to mount a defence. For all their bravery, the mages were unable to withstand the countless legions of demons, but their heroism breathed hope into the hearts of the peoples of Azeroth.#!"
+
+ guest_description.magic_good_3.desc:0 "A [GetTrait('magic_good_3').GetName( ROOT.Char )]"
```
Comment (MemeWizard) [inline localization/english/wc_general_l_english.yml:39]:
```suggestion
 guest_description.magic_good_3.desc:0 "An [GetTrait('magic_good_3').GetName( ROOT.Char )]"
```

**Review by TheGroxEmpire — APPROVED** [review]

**Pull request merged**